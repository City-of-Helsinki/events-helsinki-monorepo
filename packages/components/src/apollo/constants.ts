/**
 * The Events-graphql-proxy offers a way to ignore some custom ignorable graphql errors.
 * The 'X-Ignored-Error-Code' header is used to list the errors wanted to be ignored.
 * */
export const ignoredErrorCodesHeader = {
  'X-Ignored-Error-Code': 'UNPOPULATED_MANDATORY_DATA',
};
