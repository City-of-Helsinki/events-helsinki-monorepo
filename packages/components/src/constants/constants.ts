import { LanguageCodeEnum } from '../types';
import type { Language, AppLanguage } from '../types';

export const MAIN_CONTENT_ID = 'main-content';

export const EXTLINK = {
  EXTLINK_FACEBOOK: 'extlink_facebook',
  EXTLINK_INSTAGRAM: 'extlink_instagram',
  EXTLINK_TWITTER: 'extlink_twitter',
  EXTLINK_YOUTUBE: 'extlink_youtube',
};

export const DATE_TYPES = {
  TODAY: 'today',
  TOMORROW: 'tomorrow',
  WEEKEND: 'weekend',
  THIS_WEEK: 'this_week',
};

export const EVENT_STATUS = {
  EVENT_CANCELLED: 'EventCancelled',
  EVENT_SCHEDULED: 'EventScheduled',
};

/**
 * If logic for selecting the fallback language priority is needed,
 * then prioritize order is English, Finnish, Swedish.
 * Ref: https://helsinkisolutionoffice.atlassian.net/browse/HH-110
 */
export const APP_LANGUAGES = ['en', 'fi', 'sv'] as const;

export const APP_LANGUAGE_LABELS: Record<AppLanguage, string> = {
  en: 'English',
  fi: 'Suomi',
  sv: 'Svenska',
} as const;

export const DEFAULT_LANGUAGE: AppLanguage = 'fi';

export enum DATE_PICKER_INPUT {
  START = 'start',
  END = 'end',
}

export enum DATE_PICKER_INPUT_STATE {
  NOT_SELECTED = 'not-selected',
  START_TIME_SELECTED = 'start-time-selected',
  END_TIME_SELECTED = 'end-time-selected',
}

export const DEFAULT_SOME_IMAGE = '/images/activities_SoMe-share.jpg';

export const FEEDBACK_LINKS = {
  fi: 'https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/osallistu-ja-vaikuta/palaute',
  en: 'https://www.hel.fi/helsinki/en/administration/participate/feedback',
  sv: 'https://www.hel.fi/helsinki/sv/stad-och-forvaltning/delta/feedback',
};

export const DEFAULT_HEADER_MENU_NAME: Record<AppLanguage, string> = {
  fi:
    process.env.NEXT_PUBLIC_CMS_HEADER_MENU_NAME_FI ??
    'Events Helsinki Header FI',
  en:
    process.env.NEXT_PUBLIC_CMS_HEADER_MENU_NAME_EN ??
    'Events Helsinki Header EN',
  sv:
    process.env.NEXT_PUBLIC_CMS_HEADER_MENU_NAME_SV ??
    'Events Helsinki Header SV',
};

export const DEFAULT_FOOTER_MENU_NAME: Record<AppLanguage, string> = {
  fi:
    process.env.NEXT_PUBLIC_CMS_FOOTER_MENU_NAME_FI ??
    'Events Helsinki Footer FI',
  en:
    process.env.NEXT_PUBLIC_CMS_FOOTER_MENU_NAME_EN ??
    'Events Helsinki Footer EN',
  sv:
    process.env.NEXT_PUBLIC_CMS_FOOTER_MENU_NAME_SV ??
    'Events Helsinki Footer SV',
};

export const DEFAULT_HEADER_UNIVERSAL_BAR_MENU_NAME: Record<
  AppLanguage,
  string
> = {
  fi:
    process.env.NEXT_PUBLIC_CMS_HEADER_UNIVERSAL_BAR_MENU_NAME_FI ??
    'Events Helsinki Universal bar FI',
  en:
    process.env.NEXT_PUBLIC_CMS_HEADER_UNIVERSAL_BAR_MENU_NAME_EN ??
    'Events Helsinki Universal bar EN',
  sv:
    process.env.NEXT_PUBLIC_CMS_HEADER_UNIVERSAL_BAR_MENU_NAME_SV ??
    'Events Helsinki Universal bar SV',
};

/**
 * This is just a mock of list of Languages.
 * The react-helsinki-headless-cms needs typeof `Language[]` in a list of lanugages,
 * and at least the current language must present in the list.
 * The error page should not fetch langauges by using the Apollo client,
 * because the network error in the Apollo client is
 * usually the reason to show the error page.
 */
export const HARDCODED_LANGUAGES: Language[] = [
  {
    __typename: 'Language',
    id: 'TGFuZ3VhZ2U6Zmk=',
    locale: 'fi',
    name: 'Suomi',
    code: LanguageCodeEnum.Fi,
    slug: 'fi',
  },
  {
    __typename: 'Language',
    id: 'TGFuZ3VhZ2U6ZW4=',
    locale: 'en',
    name: 'English',
    code: LanguageCodeEnum.En,
    slug: 'en',
  },
  {
    __typename: 'Language',
    id: 'TGFuZ3VhZ2U6c3Y=',
    locale: 'sv',
    name: 'Svenska',
    code: LanguageCodeEnum.Sv,
    slug: 'sv',
  },
];
