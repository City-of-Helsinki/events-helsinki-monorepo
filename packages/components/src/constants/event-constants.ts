// Amount of the similar events

export const SIMILAR_EVENTS_AMOUNT = 8;

export const EVENT_KEYWORD_BLACK_LIST = [
  'helmet:10592', // Yleinen
  'helmet:10689', // Helsinki
  'helmet:11916', // Infonäytöt (Helsinki)
  'kulke:53', // Suomi
  'kulke:54', // Ruotsi
  'kulke:55', // Englanti
  'kulke:616', // Näkyy TV-sovelluksessa
  'kulke:671', // Normaali
  'kulke:673', // Suomi (TV)
];

export const EVENT_SOME_IMAGE = '/images/activities_SoMe-share.jpg';

export const EVENT_LOCATIONS = { INTERNET: 'helsinki:internet' };

export enum EVENT_SORT_OPTIONS {
  DURATION = 'duration',
  DURATION_DESC = '-duration',
  END_TIME = 'end_time',
  END_TIME_DESC = '-end_time',
  LAST_MODIFIED_TIME = 'last_modified_time',
  LAST_MODIFIED_TIME_DESC = '-last_modified_time',
  START_TIME = 'start_time',
  START_TIME_DESC = '-start_time',
}

export const isEventSortOption = (
  value: unknown
): value is EVENT_SORT_OPTIONS =>
  Object.values(EVENT_SORT_OPTIONS).includes(value as EVENT_SORT_OPTIONS);

export const DEFAULT_EVENT_SORT_OPTION = EVENT_SORT_OPTIONS.END_TIME;

export enum EnrolmentStatusLabel {
  enrollable = 'enrollable',
  noEnrolmentTimes = 'noEnrolmentTimes',
  full = 'full',
  enrolmentNotStartedYet = 'enrolmentNotStartedYet',
  enrolmentEnded = 'enrolmentEnded',
  queueable = 'queueable',
}

export enum EVENT_SEARCH_FILTERS {
  CATEGORIES = 'categories',
  HOBBY_TYPES = 'hobbyTypes',
  DATE_TYPES = 'dateTypes',
  DIVISIONS = 'divisions',
  END = 'end',
  HELSINKI_ONLY = 'helsinkiOnly',
  IS_FREE = 'isFree',
  KEYWORD = 'keyword',
  KEYWORD_NOT = 'keywordNot',
  ONLY_CHILDREN_EVENTS = 'onlyChildrenEvents',
  ONLY_EVENING_EVENTS = 'onlyEveningEvents',
  ONLY_REMOTE_EVENTS = 'onlyRemoteEvents',
  PAGE = 'page',
  PLACES = 'places',
  PUBLISHER = 'publisher',
  START = 'start',
  TEXT = 'xFullText',
  ONGOING = 'xOngoing',
  MIN_AGE = 'audienceMinAgeLt',
  MAX_AGE = 'audienceMaxAgeGt',
  SUITABLE = 'suitableFor',
}

/**
 * City of Helsinki's organization ID in Linked Events
 * @see https://api.hel.fi/linkedevents/v1/organization/ahjo:00001/
 */
export const CITY_OF_HELSINKI_LINKED_EVENTS_ORGANIZATION_ID =
  'ahjo:00001' as const;

/**
 * Prefixes of organization IDs under City of Helsinki root organization 'ahjo:00001',
 * @see https://api.hel.fi/linkedevents/v1/organization/ahjo:00001/
 *
 * Kultus subtree of the organization tree starts at 'kultus:0' and is excluded,
 * @see https://api.hel.fi/linkedevents/v1/organization/kultus:0/
 */
export const CITY_OF_HELSINKI_LINKED_EVENTS_ORGANIZATION_PREFIXES = [
  'ahjo',
  'helsinki',
];
