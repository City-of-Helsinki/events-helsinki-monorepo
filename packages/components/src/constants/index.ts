export * from './constants';
export * from './event-constants';
export * from './venue-constants';
