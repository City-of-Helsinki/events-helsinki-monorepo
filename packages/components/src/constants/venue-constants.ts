export const Sources = {
  TPREK: 'tprek',
  LINKED: 'linked',
} as const;

export const HELSINKI_OCD_DIVISION_ID =
  'ocd-division/country:fi/kunta:helsinki';
