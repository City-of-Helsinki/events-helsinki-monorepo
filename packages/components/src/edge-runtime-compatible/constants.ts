// TODO: For some reason middleware cannot read `'@events-helsinki/components` package without breaking the build
export const LOCALES = ['fi', 'en', 'sv'] as const; // i18n.locales.join('|');
export const DEFAULT_LANGUAGE = 'fi';
