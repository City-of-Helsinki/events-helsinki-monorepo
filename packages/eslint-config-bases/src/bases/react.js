/**
 * Opinionated config base for projects using react.
 * @see https://github.com/City-of-Helsinki/events-helsinki-monorepo/tree/main/packages/eslint-config-bases
 */

import react from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';

const reactPatterns = {
  files: ['**/*.{js,jsx,ts,tsx}'],
};

export default [
  {
    files: reactPatterns.files,
    plugins: {
      react,
    },
    settings: {
      react: {
        version: 'detect',
      },
    },
    rules: {
      ...react.configs.recommended.rules,
      // https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unknown-property.md
      'react/no-unknown-property': ['error', { ignore: ['css'] }],
      // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md
      'react/no-unescaped-entities': ['error', { forbid: ['>'] }],
      'react/prop-types': 'off',
      'react/react-in-jsx-scope': 'off',
      // Fine-tune naming convention react typescript jsx (function components)
      // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md
      '@typescript-eslint/naming-convention': [
        'warn',
        {
          selector: 'variable',
          format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
          leadingUnderscore: 'allow',
        },
        {
          selector: ['function'],
          format: ['camelCase', 'PascalCase'],
        },
        {
          selector: 'parameter',
          format: ['camelCase', 'PascalCase'],
          leadingUnderscore: 'allow',
        },
      ],
    },
  },
  {
    files: reactPatterns.files,
    plugins: {
      'react-hooks': reactHooks,
    },
    rules: {
      ...reactHooks.configs.recommended.rules,
      //   'react-hooks/rules-of-hooks': 'error',
      //   'react-hooks/exhaustive-deps': 'warn',
    },
  },
];
