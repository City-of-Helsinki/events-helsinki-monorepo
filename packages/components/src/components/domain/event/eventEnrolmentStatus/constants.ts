import { EnrolmentStatusLabel } from '../../../../constants/event-constants';

export const OPEN_ENROLMENT_STATUSES = [
  EnrolmentStatusLabel.enrollable,
  EnrolmentStatusLabel.queueable,
];
