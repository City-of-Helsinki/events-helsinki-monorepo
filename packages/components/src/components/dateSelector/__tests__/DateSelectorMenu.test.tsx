import * as React from 'react';
import { render } from '@/test-utils';
import { DATE_TYPES } from '../../../constants';
import DateSelectorMenu from '../DateSelectorMenu';

it('matches snapshot', () => {
  const { container } = render(
    <DateSelectorMenu
      dateTypes={['type1', 'type2']}
      dateTypeOptions={[
        DATE_TYPES.TODAY,
        DATE_TYPES.TOMORROW,
        DATE_TYPES.THIS_WEEK,
        DATE_TYPES.WEEKEND,
      ]}
      endDate={new Date('2019-09-31')}
      isCustomDate={false}
      isOpen={true}
      name="date"
      onChangeDateTypes={vi.fn()}
      onChangeEndDate={vi.fn()}
      onChangeStartDate={vi.fn()}
      onCloseMenu={vi.fn()}
      startDate={new Date('2019-08-01')}
      toggleIsCustomDate={vi.fn()}
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
