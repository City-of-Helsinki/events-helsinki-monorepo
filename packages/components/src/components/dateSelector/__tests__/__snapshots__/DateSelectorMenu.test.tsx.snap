// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  aria-live="polite"
  class="_dateSelectorMenu_bbd991"
  data-testid="date-selector-menu"
  role="menu"
>
  <div
    class="_checkboxWrapper_bbd991"
  >
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); --background-selected: var(--color-input-dark, --color-coat-of-arms); --background-selected-hover: var(--color-input-dark, --color-coat-of-arms-dark); --background-hover: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected: var(--color-input-dark,--color-coat-of-arms-dark); --border-color-selected-focus: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected-hover: var(--color-input-dark,--color-coat-of-arms-dark); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_today"
        name="date"
        type="checkbox"
        value="today"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_today"
      >
        Tänään
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); --background-selected: var(--color-input-dark, --color-coat-of-arms); --background-selected-hover: var(--color-input-dark, --color-coat-of-arms-dark); --background-hover: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected: var(--color-input-dark,--color-coat-of-arms-dark); --border-color-selected-focus: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected-hover: var(--color-input-dark,--color-coat-of-arms-dark); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_tomorrow"
        name="date"
        type="checkbox"
        value="tomorrow"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_tomorrow"
      >
        Huomenna
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); --background-selected: var(--color-input-dark, --color-coat-of-arms); --background-selected-hover: var(--color-input-dark, --color-coat-of-arms-dark); --background-hover: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected: var(--color-input-dark,--color-coat-of-arms-dark); --border-color-selected-focus: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected-hover: var(--color-input-dark,--color-coat-of-arms-dark); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_this_week"
        name="date"
        type="checkbox"
        value="this_week"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_this_week"
      >
        Tällä viikolla
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz"
      style="--background-unselected: var(--color-white); --background-selected: var(--color-input-dark, --color-coat-of-arms); --background-selected-hover: var(--color-input-dark, --color-coat-of-arms-dark); --background-hover: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected: var(--color-input-dark,--color-coat-of-arms-dark); --border-color-selected-focus: var(--color-input-dark, --color-coat-of-arms-dark); --border-color-selected-hover: var(--color-input-dark,--color-coat-of-arms-dark); line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_weekend"
        name="date"
        type="checkbox"
        value="weekend"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_weekend"
      >
        Viikonloppuna
      </label>
    </div>
  </div>
  <button
    class="_button_bbd991 _btnSelectDates_bbd991"
    type="button"
  >
    <svg
      aria-hidden="true"
      aria-label="calendar-plus"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M20 14V17H23V19H20V22H18V19H15V17H18V14H20ZM17 2C17.5523 2 18 2.44772 18 3V4H22V12H20V11H4V19H13V21H2V4H6V3C6 2.44772 6.44772 2 7 2C7.55228 2 8 2.44772 8 3V4H16V3C16 2.44772 16.4477 2 17 2ZM20 6H4V9H20V6Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="_buttonText_bbd991"
    >
      Valitse päivät
    </div>
    <svg
      aria-hidden="true"
      aria-label="angle-right"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M13.5 12L8.5 7L10 5.5L16.5 12L10 18.5L8.5 17L13.5 12Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="_button_bbd991 _btnBack_bbd991 _hidden_bbd991"
    type="button"
  >
    <svg
      aria-hidden="true"
      aria-label="angle-left"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10.5 12L15.5 17L14 18.5L7.5 12L14 5.5L15.5 7L10.5 12Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="_buttonText_bbd991"
    >
      Takaisin
    </div>
  </button>
</div>
`;
